### PR TITLE
Various performance and visual improvements

### PR DESCRIPTION
**Pull request by CyanDevs**
- Binary backoff for requests to not flood Jenkins master
  - Abort Jenkins job if script is stopped
  - Handle invalid HTTP GET responses
  - Retry connection errors and timeouts
  - Remove HTML tags from output
  - Upgrade to Python3

To upgrade:
- Ensure `python3` is installed
- Install `python3-pip`
- Install `docopt`, `urllib3`, and `requests` with `pip3`

Signed-off-by: Chris Yan <chrisyan@microsoft.com>

### CODE REVIEW

**Comment by BRMcLaren** [comment]
@CyanDevs 

1) Fantastic branch name,
2) Would you mind making this as a separate script? I don't have a way to test with integrating partners and am hesitant to break at this time.

**Comment by BRMcLaren** [comment]
Can you also add any additional dependencies here? https://github.com/openenclave/test-infra/blob/master/images/jenkinsoperator/Dockerfile

:D

**Comment by CyanDevs** [comment]
As requested, I've moved hack/jenkins_bootstrap.py to scripts/jenkins_bootstrap.py and added the required dependencies in the Dockerfile.

File: scripts/jenkins_bootstrap.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+        if self.encoding.lower() in ["html", "text"]:
+            self.encoding = self.encoding.title()
+        else:
+            raise ValueError(f"{self.encoding} is not a valid encoding only support 'text', 'html'")
+
+        if arguments['--parameters']:
+            try:
+                self.parameters = dict(u.split("=") for u in arguments['--parameters'][0].split(","))
+            except ValueError as err:
+                raise Exception("Your parameters should be in key=value format separated by ; for multi value i.e. x=1,b=2") from err
```
Comment (BRMcLaren) [inline scripts/jenkins_bootstrap.py:83]: "separated by ; for multi value i.e. x=1,b=2"

Wouldn't this imply the come should be a semi-colon, or the semi-colon, a comma?

Comment (CyanDevs) [reply scripts/jenkins_bootstrap.py:83]: Good catch. What did you intend to have when you wrote the original script? `;` or `,` ?

Comment (BRMcLaren) [reply scripts/jenkins_bootstrap.py:83]: I believe commas.

Comment (CyanDevs) [reply scripts/jenkins_bootstrap.py:83]: Thanks, I'll change it momentarily.

File: images/jenkinsoperator/Dockerfile [hunk 2]
```diff
@@ -17,9 +17,11 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     pkg-config \
     procps \
     python \
+    python3 \
     python-dev \
     python-openssl \
     python-pip \
+    python3-pip \
```
Comment (BRMcLaren) [inline images/jenkinsoperator/Dockerfile:24]: Were you able to build this image locally? When I tried I needed to add the pip3 set up tools.

Comment (CyanDevs) [reply images/jenkinsoperator/Dockerfile:24]: python3-pip installs fine for me. However, the pip3 update failed and that might've been what you were seeing. I've added changes to the upgrade line below to fix it.

**Pull request merged**